cmd/promtool/main.go

```diff
@@ -172,6 +175,24 @@ func main() {
 		p = &promqlPrinter{}
 	}
 
+	var queryOpts promql.LazyLoaderOpts
+	for _, f := range *featureList {
```

Comment:
Should we add a comment in cmd/prometheus to update here when new flags are added?

Reply:
I do wonder if rather than adding this LazyLoaderOpts it would make sense to move the features out of EngineOpts into a struct that is included in both? Potentially even move the string names of the features into the promql package...

Reply:
I like the idea of moving them all struct. It would be even better if we also created a singular parsing function that could be used in both `main`s, that way we never have to worry about updating this. Prometheus `main` parses other disabled features that don't relate to PromQL though, so we'd have to keep that in mind.

Reply:
Yeah, maybe keep the list handling in each main, then for the promql features something like:

in package promql:
```
type Features struct {
...
}

func (f *Features) Enable(name string) bool {
  switch name {
  case "promql-at-modifier":
      f.EnablePromQLAt = true
      return true
  default:
    return false
}
```

and then each main can also support features outside that set if it likes.

Reply:
But merged, that seems like a useful follow-up.